### PR TITLE
DOC: improve and clean up *Spline docstrings in fitpack2.py

### LINKED ISSUE #11756: Improve and fix *Spline docstrings and code
During the investigation of the issue #9718 I've discovered a few possibilities for both improvements and fixes in the documentation of different *Spline classes in 
https://github.com/scipy/scipy/blob/master/scipy/interpolate/fitpack2.py
I create this issue mainly because it seems to me that it'll be (and must be) more general than #9718
The things below will be mostly about `RectBivariateSpline` class

Things that came into mind:

- In `BivariateSpline` class, as far as I understand,  `RectBivariateSpline` should be added to the end of the sentence, because it subclasses `BivariateSpline`
https://github.com/scipy/scipy/blob/b818978f3580e05a3885006cae337f28bd2aaf7d/scipy/interpolate/fitpack2.py#L931-L933
- Docstring of `RectBivariateSpline` containes a few more inconsistencies: e.g. `tx, ty` in here: https://github.com/scipy/scipy/blob/b818978f3580e05a3885006cae337f28bd2aaf7d/scipy/interpolate/fitpack2.py#L1149-L1152
 although they do not appear in `init`:
https://github.com/scipy/scipy/blob/b818978f3580e05a3885006cae337f28bd2aaf7d/scipy/interpolate/fitpack2.py#L1169
- I have a feeling that "See Also" sections in different Spline's might be more... broad? Consistent? There's not so many of that classes, so why not to mention them all, or at least ensure that if `A -> B` and `A -> C` (-> means subclassing), then `B` mentions `C` in "See Also" and vice versa?
- I've got a feeling that two last checks of these 4 are redundant, but I'm not sure
https://github.com/scipy/scipy/blob/b818978f3580e05a3885006cae337f28bd2aaf7d/scipy/interpolate/fitpack2.py#L1171-L1178
- Problem with #9718 remains for now, but I'm working on it

If this issue "makes sense" to some of the members, I suggest closing #9718 in favor of this one.

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
fix #11756 

#### What does this implement/fix?

I improved and cleaned up *Spline docstrings in fitpack2.py to fix some issues reported in #11756.

I added.

I think this is a doc bug, this function does not use `tx`, `ty`. 
`bbox` is just use min-max of x and y based on this code:
https://github.com/scipy/scipy/blob/fd6f6d670909681127be0e545bc1a116d6b74a79/scipy/interpolate/src/fitpack.pyf#L465-L468
So, I fixed it.

##### Issue 3

> I have a feeling that "See Also" sections in different Spline's might be more... broad? Consistent? There's not so many of that classes, so why not to mention them all, or at least ensure that if A -> B and A -> C (-> means subclassing), then B mentions C in "See Also" and vice versa?

I cleaned up docstrings of some *Spline. These uses a same "See Also" contents.

##### Issue 4

> I've got a feeling that two last checks of these 4 are redundant, but I'm not sure 

Already, these redundant input check were removed in the master code.

##### Issue 5

> The docstring of the smoothing factor s of the class RectBivariateSpline mentions the vector w as part of the estimation condition:

I think this is a doc bug, this function does not use `w` discussed in #9718. So I removed it. Also I updated estimation function docstring. Because there are two 's', first one is spline function. I clarified it.

### CODE REVIEW

File: scipy/interpolate/fitpack2.py [hunk 1]
```diff
@@ -1644,8 +1738,23 @@ class RectSphereBivariateSpline(SphereBivariateSpline):
 
     See Also
     --------
-    RectBivariateSpline : bivariate spline approximation over a rectangular
-        mesh
+    BivariateSpline :
+        to create a bivariate spline on a plane grid
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:1742]: Looking up the docs, the docstring of BivariateSpline, https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.BivariateSpline.html, says explicitly 

> Base class for bivariate splines. ... This class is meant to be subclassed, not instantiated directly.

IMO, it'd be better to have this in the SeeAlso sections instead, here and elsewehere.

File: scipy/interpolate/fitpack2.py [hunk 2]
```diff
@@ -1644,8 +1738,23 @@ class RectSphereBivariateSpline(SphereBivariateSpline):
...
+    UnivariateSpline :
+        a similar class for univariate spline interpolation
+    SmoothBivariateSpline :
+        to create a bivariate spline through the given points
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:1746]: Maybe reword? Reading this line leaves me as confused as before, and having to look up the docs. Maybe just say it's a smoothing bivariate spline and leave it at that (more details are available in the SmoothBivariareSpline docstring; the latter may use an example, but that's separate).

File: scipy/interpolate/fitpack2.py [hunk 3]
```diff
@@ -1644,8 +1738,23 @@ class RectSphereBivariateSpline(SphereBivariateSpline):
...
+    LSQBivariateSpline :
+        to create a bivariate spline using weighted least-squares fitting
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:1748]: I'm also not sure about this "to create a ..." clause. Maybe just shorten to "A bivariate spline using ...". Feel free to igore though, it's ESL here :-).

File: scipy/interpolate/fitpack2.py [hunk 4]
```diff
@@ -588,9 +602,9 @@ class InterpolatedUnivariateSpline(UnivariateSpline):
     UnivariateSpline : Superclass -- allows knots to be selected by a
         smoothing condition
     LSQUnivariateSpline : spline for which knots are user-selected
+    BivariateSpline : A similar class for two-dimensional spline interpolation
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:605]: Nit: `UnivariateSpline` is meant to be instantiated while `BivariateSpline` is a base class. So maybe, just maybe, link to LSQBivariateSpline and SmoothBivariateSpline instead.

File: scipy/interpolate/fitpack2.py [hunk 5]
```diff
@@ -588,9 +602,9 @@ class InterpolatedUnivariateSpline(UnivariateSpline):
...
     splrep : An older, non object-oriented wrapping of FITPACK
     splev, sproot, splint, spalde
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:607]: as above.

**Review by ev-br — CHANGES_REQUESTED** [review]
Is definitely an improvement, but I'd ask for some more inline.

**Comment by stefanv** [comment]
Thanks for taking a look at this, @AtsushiSakai

File: scipy/interpolate/fitpack2.py [hunk 6]
```diff
@@ -120,12 +120,26 @@ class UnivariateSpline(object):
...
+        to create a bivariate spline in spherical coordinates using
+        weighted least-squares fitting
+    RectBivariateSpline :
+        to create a bivariate spline over a rectangular mesh.
+    bisplrep : older wrapping of FITPACK
+    bisplev : older wrapping of FITPACK
     InterpolatedUnivariateSpline : Subclass with smoothing forced to 0
-    LSQUnivariateSpline : Subclass in which knots are user-selected instead of
-        being set by smoothing condition
     splrep : An older, non object-oriented wrapping of FITPACK
```
[the remark this answers is not in the document]
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: OK. I agree your comment. I fixed it. PTAL.

File: scipy/interpolate/fitpack2.py [hunk 7]
```diff
@@ -1057,10 +1073,23 @@ class SmoothBivariateSpline(BivariateSpline):
...
+        to create a bivariate spline over a rectangular mesh on a sphere
+    SmoothSphereBivariateSpline :
+        to create a smooth bivariate spline in spherical coordinates
+    LSQSphereBivariateSpline :
+        to create a bivariate spline in spherical coordinates using
+        weighted least-squares fitting
+    RectBivariateSpline :
+        to create a bivariate spline over a rectangular mesh.
+    bisplrep : older wrapping of FITPACK
+    bisplev : older wrapping of FITPACK
```
[the remark this answers is not in the document]
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I fixed it.

File: scipy/interpolate/fitpack2.py (hunk 1 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I use "a base class for bivariate splines" for one line comment in SeeAlso. PTAL.

File: scipy/interpolate/fitpack2.py (hunk 2 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I clean up the comment.

File: scipy/interpolate/fitpack2.py (hunk 4 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I think this is not needed in SeeAlso because it is one of the UnivariateSpline function. So I removed it.

File: scipy/interpolate/fitpack2.py (hunk 5 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I fixed it.

File: scipy/interpolate/fitpack2.py (hunk 3 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: I totally agree your comment. So, I shortened the comment.

File: scipy/interpolate/fitpack2.py [hunk 8]
```diff
@@ -120,12 +120,38 @@ class UnivariateSpline(object):
...
+        a bivariate spline over a rectangular mesh on a sphere
+    SmoothSphereBivariateSpline :
+        a smooth bivariate spline in spherical coordinates
+    LSQSphereBivariateSpline :
+        a bivariate spline in spherical coordinates using weighted
+        least-squares fitting
+    RectBivariateSpline :
+        a bivariate spline over a rectangular mesh
+    InterpolatedUnivariateSpline :
+        a interpolating unicariate spline for a given set of data points.
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:139]: Typo uni*v*ariate

File: scipy/interpolate/fitpack2.py [hunk 9]
```diff
@@ -120,12 +120,38 @@ class UnivariateSpline(object):
...
-    InterpolatedUnivariateSpline : Subclass with smoothing forced to 0
-    LSQUnivariateSpline : Subclass in which knots are user-selected instead of
-        being set by smoothing condition
-    splrep : An older, non object-oriented wrapping of FITPACK
-    splev, sproot, splint, spalde
-    BivariateSpline : A similar class for two-dimensional spline interpolation
+    BivariateSpline :
+        a base class for bivariate splines.
+    SmoothBivariateSpline :
+        a smooth bivariate spline through the given points
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py:126]: I think the term is smooth*ing* spline (all of them are smooth, but this one does smoothing of the noise in the input data)

**Review by ev-br — COMMENTED** [review]
Two more minor comments, sorry I missed them in the first review

File: scipy/interpolate/fitpack2.py (hunk 8 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: Thank you for catching, I fixed it.

File: scipy/interpolate/fitpack2.py (hunk 9 above)
Comment (AtsushiSakai) [reply scipy/interpolate/fitpack2.py]: Agree. I fixed it.

**Comment by ev-br** [comment]
Thanks @AtsushiSakai, merging
